src/ImageService.Broker/Consumers/CreateImagesConsumer.cs

```diff
@@ -0,0 +1,194 @@
+using LT.DigitalOffice.ImageService.Mappers.Db.Interfaces;
+using LT.DigitalOffice.ImageService.Mappers.Helpers.Interfaces;
+using LT.DigitalOffice.ImageService.Models.Db;
+using LT.DigitalOffice.Kernel.Broker;
+using LT.DigitalOffice.Models.Broker.Enums;
+using LT.DigitalOffice.Models.Broker.Models;
+using LT.DigitalOffice.Models.Broker.Requests.Image;
+using LT.DigitalOffice.Models.Broker.Responses.Image;
+using MassTransit;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.ImageService.Broker.Consumers
+{
+    public class CreateImagesConsumer : IConsumer<ICreateImageRequest>
+    {
+        private readonly IImageUserRepository _imageUserRepository;
+        private readonly IImageProjectRepository _imageProjectRepository;
+        private readonly IImageNewsRepository _imageNewsRepository;
+        private readonly IDbImageUserMapper _dbImageUserMapper;
+        private readonly IDbImageNewsMapper _dbImageNewsMapper;
+        private readonly IDbImageProjectMapper _dbImageProjectMapper;
+        private readonly IResizeImageHelper _resizeHelper;
+
+        public CreateImagesConsumer(
+            IImageUserRepository imageUserRepository,
+            IImageProjectRepository imageProjectRepository,
+            IImageNewsRepository imageNewsRepository,
+            IDbImageUserMapper dbImageUserMapper,
+            IDbImageNewsMapper dbImageNewsMapper,
+            IDbImageProjectMapper dbImageProjectMapper,
+            IResizeImageHelper resizeHelper)
+        {
+            _imageUserRepository = imageUserRepository;
+            _imageProjectRepository = imageProjectRepository;
+            _imageNewsRepository = imageNewsRepository;
+            _dbImageUserMapper = dbImageUserMapper;
+            _dbImageNewsMapper = dbImageNewsMapper;
+            _dbImageProjectMapper = dbImageProjectMapper;
+            _resizeHelper = resizeHelper;
+        }
+
+        public async Task Consume(ConsumeContext<ICreateImageRequest> context)
+        {
+            object response;
+
+            switch (context.Message.ImageSource)
+            {
+                case ImageSource.User:
+                    response = OperationResultWrapper.CreateResponse(CreateUserImages, context.Message);
+                    break;
+
+                case ImageSource.Project:
+                    response = OperationResultWrapper.CreateResponse(CreateProjectImages, context.Message);
+                    break;
+
+                case ImageSource.News:
+                    response = OperationResultWrapper.CreateResponse(CreateNewsImages, context.Message);
+                    break;
+
+                default:
+                    response = null;
+                    break;
+            }
+
+            await context.RespondAsync<IOperationResult<ICreateImagesResponse>>(response);
+        }
+
+        private object CreateUserImages(ICreateImageRequest request)
```

Comment:
You can do one method instead of four.

Reply:
If i put it in the case body, it will be wery big method. We can't also use one method for each entity because of entity repositories are different
Don't know how to change it to decrase its size

Reply:
delegate object f1(ICreateImageRequest req);
func f2(f1, msgType)

response = f2(CreateMessageImages or CreateUserImages or smg else, context.Message)

mb like that?